stdlib/Random/src/Random.jl

```diff
@@ -248,6 +249,7 @@ function __init__()
         Base.showerror_nostdio(ex,
             "WARNING: Error during initialization of module Random")
     end
+    ThreadRNG[] = randjump(GLOBAL_RNG, big(10)^20, Threads.nthreads()) 
```

Comment:
Note that this must be done on multiple threads instead of only one one thread.

Reply:
Do you mean implementing something like `resize_nthreads!()` but that assigns to each buffer a different generator?

Reply:
No I mean the engine has to be allocated on different threads. See the recent discourse thread about this. (On my phone, that+ to past link)

Reply:
Yeah I had [this discussion](https://discourse.julialang.org/t/thread-local-storage-in-julia/11765) with @stevengj about placing TLS generators.
The problem is that 
1) Using `task_local_storage` is impractical as they often get dereferenced.
2) Writing a interface to `pthread_setspecifc()` can be useless, as retrieving variables there can be slower than accessing an array. Also it would bring the problem of managing the keys at runtime.

Reply:
No it has NOTHING to do with task_local_storage or  pthread tls API. You just need to "allocated on different threads", i.e. let the thread allocate, that's it.

> Where also it would not probably benefit us for performance

Oh, it'll absolutely kill you. Again, see the benchmarks in the discourse thread.

Reply:
Ehm the discussion above was started by myself, and I did those benchmark (sorry I'm Gaetano :D ). If you refer to another one, please link it to me.

The problem is that the definition of `trand()` (that present in the benchmarks linked above) is done at runtime, and the ThreadRNG const array was also instantiated at runtime. That's why there I had the opportunity to let a thread allocate the array of `MersenneTwister`

Reply:
> Ehm the discussion above was started by myself

I never mention "discussion above", I said "discourse thread" which I didn't link since I was on my phone as [explained above](https://github.com/JuliaLang/julia/pull/27950#discussion_r201020033).
I don't think that thread is that hard to find. It's currently the second result when searching for "rand" and "thread" (sorted with date). https://discourse.julialang.org/search?q=rand%20thread%20order%3Alatest, the direct link is https://discourse.julialang.org/t/poor-performance-on-cluster-multithreading

> ThreadRNG const array was also instantiated at runtime.

And speaking of that, I don't really understand why it has to be a ref of vector. Vector is mutable already.

Reply:
Sorry, I thought about a previous thread.
OK I'm going to write an helper function to do that. Thanks for your support!